subprojects/docs/src/docs/userguide/plugin_reference.adoc

```diff
@@ -25,6 +25,9 @@ Provides support for building any type of Java project.
 <<java_library_plugin.adoc#,Java Library>>::
 Provides support for building a Java library.
 
+<<java_platform_plugin.adoc#,Java Platform>>::
+Provides support for building a Java platform.
```

Comment:
This is the minimal version. I wonder if we should add a bit more content to differentiate the different Java plugins without having the user click through.